go/libraries/doltcore/sqle/enginetest/dolt_queries_rebase.go

```diff
@@ -0,0 +1,702 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enginetest
+
+import (
+	"github.com/dolthub/go-mysql-server/enginetest/queries"
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	gmstypes "github.com/dolthub/go-mysql-server/sql/types"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/rebase"
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dprocedures"
+)
+
+var DoltRebaseScriptTests = []queries.ScriptTest{
+	{
+		Name:        "dolt_rebase errors: basic errors",
+		SetUpScript: []string{},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:          "call dolt_rebase('--abort');",
+				ExpectedErrStr: "no rebase in progress",
+			}, {
+				Query:          "call dolt_rebase('--continue');",
+				ExpectedErrStr: "no rebase in progress",
+			}, {
+				Query:          "call dolt_rebase('main');",
+				ExpectedErrStr: "non-interactive rebases not currently supported",
+			}, {
+				Query:          "call dolt_rebase('-i');",
+				ExpectedErrStr: "not enough args",
+			}, {
+				Query:          "call dolt_rebase('-i', 'main1', 'main2');",
+				ExpectedErrStr: "rebase takes at most one positional argument.",
+			}, {
+				Query:          "call dolt_rebase('--abrot');",
+				ExpectedErrStr: "error: unknown option `abrot'",
+			}, {
+				Query:          "call dolt_rebase('-i', 'doesnotexist');",
+				ExpectedErrStr: "branch not found: doesnotexist",
+			},
+		},
+	},
+	{
+		Name: "dolt_rebase errors: working set not clean",
+		SetUpScript: []string{
+			"create table t (pk int primary key);",
+			"call dolt_commit('-Am', 'creating table t');",
+			"insert into t values (0);",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:          "call dolt_rebase('-i', 'main');",
+				ExpectedErrStr: dprocedures.ErrRebaseUncommittedChanges.Error(),
+			},
+			{
+				Query:    "call dolt_add('t');",
+				Expected: []sql.Row{{0}},
+			},
+			{
+				Query:          "call dolt_rebase('-i', 'main');",
+				ExpectedErrStr: dprocedures.ErrRebaseUncommittedChanges.Error(),
+			},
+		},
+	},
+	{
+		SkipPrepared: true,
+		Name:         "dolt_rebase errors: no database selected",
+		SetUpScript: []string{
+			"create database temp;",
+			"use temp;",
+			"drop database temp;",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:    "select database();",
+				Expected: []sql.Row{{nil}},
+			},
+			{
+				// TODO: This test isn't working because AssertErr is called and currently always
```

Comment:
Probably worth writing a stand-alone test of this behavior or else fixing the gms side

Reply:
Fixed in GMS and unskipped this test in bb65111be4a8edd082602e3f8539768a01f2deba